info-compiler/src/test/java/com/google/sps/WebCrawlerTest.java

```diff
@@ -12,12 +12,215 @@
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
 
-  public static void main(String[] args) {
-    WebCrawler myWebCrawler = new WebCrawler();
-    myWebCrawler.scrapeHtml("https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html");
+/**
+ * A tester for web scrawler's news article compilation process, excluding content extraction,
+ * content processing and relevancy checking which are defined in standalone classes.
+ */
+@RunWith(JUnit4.class)
+public final class WebCrawlerTest {
+  private final static String CANDIDATE_NAME = "Alexandria Ocasio-Cortez";
+  private final static String CANDIDATE_ID = "1";
+  private final static String VALID_PROTOCOL = "https";
+  private final static String INVALID_PROTOCOL = "htttttps";
+  private final static String VALID_HOST = "www.cnn.com";
+  private final static String INVALID_HOST = "www.cnn.cooooom";
+  private final static String VALID_URL =
+    "https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html";
+  private final static String VALID_URL_ROBOTS_TXT =
+    "https://www.cnn.com/robots.txt";
+  private final static NewsArticle EXPECTED_NEWS_ARTICLE =
+    new NewsArticle(
+        "AOC wins NY Democratic primary against Michelle Caruso-Cabrera, CNN projects - " +
+        "CNNPolitics",
+        VALID_URL,
+        "Washington (CNN)Freshman Democratic Rep. Alexandria Ocasio-Cortez will defeat former " +
+        "longtime CNBC correspondent and anchor Michelle Caruso-Cabrera in a Democratic " +
+        "primary election on Tuesday for New York's 14th Congressional District, CNN projects.");
+  private final static String EMPTY_ABBREVIATED_CONTENT = "";
+  private final static int DELAY = 1;
+
+  private static WebCrawler webCrawler;
+  private static LocalDatastoreHelper datastoreHelper;
+  private static Datastore datastore;
+
+  @BeforeClass
+  public static void initialize() throws InterruptedException, IOException {
+    datastoreHelper = LocalDatastoreHelper.create();
+    datastoreHelper.start();
+    datastore = datastoreHelper.getOptions().getService();
+    webCrawler = new WebCrawler(datastore);
+  }
+
+  /**
+   * Resets the internal state of the Datastore emulator and then {@code datastore}. Also resets
+   * {@code webCrawler}. We choose to reset, instead of creating/destroying the Datastore emulator
+   * at each test is because {@code datastoreHelper.stop()} sometimes generates a {@code
+   * java.net.ConnectException}, when making HTTP requests. Hence we try to limit the number of
+   * times {@code datastoreHelper} is created/destroyed.
+   */
+  @Before
+  public void resetDatastore() throws IOException {
+    datastoreHelper.reset();
+    datastore = datastoreHelper.getOptions().getService();
+    webCrawler = new WebCrawler(datastore);
```

Comment:
Maybe pass in a newsContentExtractor and/or relevancyChecker so you can mock them?